lib/components/tag/tag.dart

```diff
@@ -0,0 +1,213 @@
+import 'dart:html';
+import 'package:katikati_ui_lib/components/button/button.dart';
+import 'package:katikati_ui_lib/components/logger.dart';
+
+var logger = Logger('tag.dart');
+
+enum TagStyle {
+  None,
+  Green,
+  Yellow,
+  Red,
+  Important,
+}
+
+class TagView {
+  SpanElement renderElement;
+  SpanElement _tagText;
+  SpanElement _tagActions;
+
+  String _text;
+  String _tagId;
+
+  bool _selectable;
+  bool _editable;
+  bool _editableOnAdd;
+  bool _removable;
+  bool _acceptable;
+  bool _suggested;
+
+  TagStyle _tagStyle;
+
+  ButtonElement _editButton;
+  ButtonElement _deleteButton;
+  ButtonElement _confirmButton;
+  ButtonElement _cancelButton;
+  ButtonElement _acceptButton;
+
+  // some default callbacks, so we can use some behaviours like reset without implementing the raw onclicks
+  void Function(String) onEdit = (_) {};
+  void Function() onDelete = () {};
+  void Function() onSelect = () {};
+  void Function() onAccept = () {};
+
+  TagView(this._text, this._tagId,
+      {String groupId,
+      bool selectable = true,
+      bool editable = false,
+      bool editableOnAdd = false,
+      bool removable = false,
+      bool acceptable = false,
+      bool suggested = false,
+      TagStyle tagStyle = TagStyle.None,
+      bool disableDoubleClickEdit = false}) {
+    _selectable = selectable;
+    _editable = editable;
+    _editableOnAdd = editableOnAdd;
+    _removable = removable;
+    _acceptable = acceptable;
+    _suggested = suggested;
+
+    renderElement = SpanElement()
+      ..dataset['id'] = _tagId
+      ..dataset['group-id'] = groupId ?? ''
+      ..classes.add('tag');
+
+    _tagText = SpanElement()
+      ..classes.add('tag-text')
+      ..dataset['placeholder'] = "untitled tag";
+    _tagActions = SpanElement()..classes.add('tag-actions');
+
+    _tagText.innerText = this._text;
+    if (suggested) {
+      var suggestedIcon = SpanElement()..className = 'fas fa-robot tag-suggested__icon';
+      _tagText.append(suggestedIcon);
+      _tagText.classes.toggle('tag-text--suggested', true);
+    }
+
+    updateStyle(tagStyle);
+
+    if (_selectable) {
+      _tagText.onClick.listen((_) {
+        onSelect();
+      });
+    }
+
+    if (_acceptable) {
+      _acceptButton = Button(ButtonType.confirm, onClick: (_) {
```

Comment:
I think it would be preferable to have `_acceptButton` be of type `Button` rather than `ButtonElement` - if you need in the future to use some functionality that the `Button` provides, you will just have access to it, whereas if you only store the underlying HTML element, you've lost that. (comment applies to all the other buttons below)